src/platform/backends/qemu/qemu_virtual_machine_factory.cpp

```diff
@@ -277,6 +274,17 @@ void mp::QemuVirtualMachineFactory::hypervisor_health_check()
 {
     mp::backend::check_for_kvm_support();
     mp::backend::check_if_kvm_is_in_use();
+
+    try
+    {
+        dnsmasq_server.check_dnsmasq_running();
+    }
+    catch (const std::exception&)
+    {
+        // Try starting dnsmasq if it's not running
+        dnsmasq_server.start_dnsmasq();
+        dnsmasq_server.check_dnsmasq_running();
+    }
```

Comment:
Do we need this here? This removes the RAII-ness of DNSMasqServer, where if it can be created, we can trust dnsmasq is running. 

Could not DNSMasqServer's constructor do this, and throw if it cannot create the process? And then these methods can be made private.

Reply:
`DNSMasqServer` is currently and has been a non-pointer class member, so RAII doesn't really apply to that.  That said, we could certainly move it to be a `std::unique_ptr` and only need to check if it's a `nullptr` or not, and if it is, then try to initialize it within a `try`/`catch` block.  I'll look into that.

Reply:
So I thought about this some more and I don't think assuming that if the `DNSMasqServer` ctor doesn't throw, we can trust `dnsmasq` is running is enough.  What happens if `dnsmasq` dies after construction?  We need some way to query that `dnsmasq` is alive which is the reason for the `check_dnsmasq_running()` method.  And then we need a method for trying to start it if it's not running.  I suppose I could wrap that up in a single method, but I think it's better to leave `DNSMasqServer` as a non-pointer object and use `check_dnsmasq_running()` for querying when needed.

Reply:
Before you explained why you daemonized dnsmasq, I had thought we could reliably manage the dnsmasq process with this DNSMasqServer class. With the way you're doing it here, you can't really.